### PR TITLE
docs: Added -rfakeroot to dpkg-buildpackage and simplified

### PR DESCRIPTION
**Pull request by smoe**
This addresses (at least as a first shot) the issues @rodw-au experienced in https://github.com/LinuxCNC/linuxcnc/issues/1363 . Many thanks to Rod for his report.

### CODE REVIEW

**Comment by rodw-au** [comment]
Thank you for fixing this. Much appreciated.

File: docs/src/code/building-linuxcnc.adoc [hunk 1]
```diff
@@ -291,6 +292,14 @@ platform to build for.  The normal values for this argument are:
     You must be running an RTAI kernel on your build machine for this
     to work!
 
+`--help`::
+
+    presets a very short summary of extra options
```
Comment (SebKuzminsky) [inline docs/src/code/building-linuxcnc.adoc:297]: Type - should be "Presents".

**Review by SebKuzminsky — CHANGES_REQUESTED** [review]
Typo.

**Comment by SebKuzminsky** [comment]
The current (ie before this PR) build instructions work for me without `-rfakeroot` on latest master, b58e0b62ca545555d7.  There must be something about this problem that i'm not understanding.

**Comment by rodw-au** [comment]
Others reported the same and I was able to create the debs without -rfakeroot on my chromebook that had not been building the debs previously. I had been building the debs on a regular basis on Debian 11 on my production machine for about a month until it failed with some kind of permissions issue when I raised issue  #1363. I suspect the issue may have been caused by a conflict with permissions on files that had been created in a previous build.
I  did observe a number of commits in relation to the .deb build process in the few days before I observed the issue so I assumed something had changed. I was able to build the debs when logged in as root.

**Comment by rodw-au** [comment]
I wonder if its because my user and root have different passwords?
I usually always make the root password and user password identical but I didn't on this build.

**Comment by SebKuzminsky** [comment]
> I wonder if its because my user and root have different passwords?

That shouldn't matter.

**Comment by phillc54** [comment]
I haven't noticed any issues without -rfakeroot either.

**Comment by smoe** [comment]
That is weird. Could it be the version of Debian that makes the difference? I am building on Debian unstable, which should be mostly equivalent to bookworm (version 12). I am not sure about what is the right thing to do, but since -rfakeroot with uttermost confidence does not do any harm (that is how packages were always built not too long ago) and with just a bit less of a confidence someone else will go through the same pain like @rodw-au , I suggest to just add it.

File: docs/src/code/building-linuxcnc.adoc (hunk 1 above)
Comment (smoe) [reply docs/src/code/building-linuxcnc.adoc]: Done. It unfortunately took me another iteration to recognize that the sentence should start with a capital letter. Also  fixed for the following section.

**Comment by smoe** [comment]
@petterreinholdtsen , does this ring a bell?

**Comment by phillc54** [comment]
I build on both MX-Linux19 (a Debian10 spinoff) and MX-Linux21 (a Debian11 spinoff)

**Comment by petterreinholdtsen** [comment]
[Steffen Möller]
> @petterreinholdtsen , does this ring a bell?

I suspect the handing of Rules-Requires-Root, which is fairly recent in
Debian, might be related.  Personally I use 'debuild' from devscripts,
which insert fakeroot when needed, so I never have to add fakeroot
settings manually.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by smoe** [comment]
>> [Steffen Möller]
>> @petterreinholdtsen , does this ring a bell?

> I suspect the handing of Rules-Requires-Root, which is fairly recent in Debian, might be related. Personally I use 'debuild' from devscripts, which insert fakeroot when needed, so I never have to add fakeroot settings manually.

@SebKuzminsky , I suggest to then go for the -rfakeroot.

**Comment by SebKuzminsky** [comment]
It's a bit hard to tell from the bug report since it does not specify which git commits built and which failed, but my hunch is that @rodw-au 's build problem stems not from problems with the build instructions or with the contents of the git repo, but from root-owned files being accidentally created and later not being able to remove/clean those files as a non-root user.

I don't want to train our users to add `fakeroot` (or worse, `su`) where it's not needed.  So the updates to the docs that add `-r fakeroot` should be removed imo.

I *do* like the updates to the build docs that simplify running `debian/configure` instead of `cd debian; ./configure; cd ..`.  Please leave those in.

With that change (and maybe a change to the title of the PR) I think it will be ready to merge.

**Comment by petterreinholdtsen** [comment]
Note, this patch is partly made obsolete by #1835 .  I strongly recommend rewriting the documentation to use 'debuild' instead of introducing -rfakeroot, as it is a more high level tool doing more quality checking automatically.

**Comment by smoe** [comment]
Too outdated to chase this all up.

**Pull request closed without merging**